libs/libvtrutil/src/vtr_random.cpp

```diff
@@ -1,91 +1,67 @@
-#include <cstddef>
 
 #include "vtr_random.h"
+#include "specrand.h"
 #include "vtr_util.h"
 #include "vtr_error.h"
-#include "specrand.h"
-
-#define CHECK_RAND
 
 namespace vtr {
-/* Portable random number generator defined below.  Taken from ANSI C by  *
- * K & R.  Not a great generator, but fast, and good enough for my needs. */
 
-constexpr size_t IA = 1103515245u;
-constexpr size_t IC = 12345u;
-constexpr size_t IM = 2147483648u;
-
-static RandState random_state = 0;
-
-/**
- * @brief The pseudo-random number generator is initialized using the argument passed as seed.
- */
-void srandom(int seed) {
-    random_state = (unsigned int)seed;
-#ifdef SPEC_CPU
-    /* SPEC CPU requires a different random number generator */
-    spec_init_genrand((unsigned long)seed);
-#endif
+RandomNumberGenerator::RandomNumberGenerator(int seed) {
+    random_state_ = (unsigned int)seed;
 }
 
-/* returns the random_state value */
-RandState get_random_state() {
-    return random_state;
+RandomNumberGenerator::RandomNumberGenerator()
+    : RandomNumberGenerator(0) {}
+
+void RandomNumberGenerator::srandom(int seed) {
+    random_state_ = (unsigned int)seed;
 }
 
-int irand(int imax, RandState& state) {
-#ifdef SPEC_CPU
-    /* SPEC CPU requires a different random number generator */
-    return (int)(spec_genrand_int31() % (imax + 1));
-#else
-    /* Creates a random integer between 0 and imax, inclusive.  i.e. [0..imax] */
+int RandomNumberGenerator::irand(int imax) {
+    // Creates a random integer between 0 and imax, inclusive.  i.e. [0..imax]
     int ival;
 
-    /* state = (state * IA + IC) % IM; */
-    state = state * IA + IC; /* Use overflow to wrap */
-    ival = state & (IM - 1); /* Modulus */
+    // state = (state * IA + IC) % IM;
+    random_state_ = random_state_ * IA + IC; // Use overflow to wrap
+    ival = random_state_ & (IM - 1); // Modulus
```

Comment:
Could/should we just use the SPEC random number generator all the time?  I put in the original random number generator, but if the SPEC one has better randomness and isn't noticeably slower we could just make these routines call it.  

I don't have a really strong opinion about this, but maybe we can simplify a code path.

Reply:
I am afraid that if we use SPEC random generator, we need to update golden results for a large number of tests. We can explore the performance/quality trade-off of these two random generators later.